Tests/Primer/Modules/PollingModuleTests.swift

```diff
@@ -97,4 +139,29 @@ class PollingModuleTests: XCTestCase {
 
         wait(for: [expectation], timeout: 30.0)
     }
+
+    func test_polling_failure_due_to_client_token_async()async throws {
+        let mockApiClient = MockPrimerAPIClient()
+        mockApiClient.pollingResults = [
+            (PollingResponse(status: .pending, id: "0", source: "src"), nil),
+            (PollingResponse(status: .pending, id: "0", source: "src"), nil),
+            (PollingResponse(status: .complete, id: "0", source: "src"), nil)
+        ]
+
+        PollingModule.apiClient = mockApiClient
+        AppState.current.clientToken = nil
+
+        let pollingModule = PollingModule(url: URL(string: "https://random.url")!)
+
+        do {
+            _ = try await pollingModule.start()
+            XCTAssert(false, "Polling succeeded, but it should fail with error .invalidClientToken")
+        } catch {
+            if let primerErr = error as? PrimerError, case .invalidClientToken = primerErr {
```

Comment:
suggestion: this might be more appropriate as a guard:
```suggestion
             guard let primerErr = error as? PrimerError, case .invalidClientToken = primerErr else {
                 XCTAssert(false, "Polling failed with error \(error.localizedDescription), but it should fail with error .invalidClientToken")
```